pkg/cosign/cosign_test.go

```diff
@@ -96,6 +96,24 @@ func TestCosignPayload(t *testing.T) {
 	assert.Equal(t, d2, "sha256:ee53528c4e3c723945cf870d73702b76135955a218dd7497bf344aa73ebb4227")
 }
 
+func TestCosignInvalidSignatureAlgorithm(t *testing.T) {
```

Comment:
I did not figure out how to do a positive test because I would need to have the correct public key.

Reply:
Does cosign cli allows us to specify the signing algorithm while signing? I think we will have to use openssl to sign the image and then upload the .sig manually to the image.

I think its fine to use any new key-cert pair from cosign generate-key-pair, those other signatures were made a while ago by different people, it will be easier to generate a new key 😅.

Reply:
@honnix, since we are just taking the value and passing it to cosign, I don't think we have to add any new positive tests. 
It would be nice if you can also add a check to make sure the value is found in the signatureAlgorithmMap and throw and error if the value is invalid.

Reply:
Sgtm.

Reply:
I did an explicit check for the provided algorithm in a latter commit.